mypy/checker.py

```diff
@@ -2674,6 +2684,52 @@ def check_member_assignment(self, instance_type: Type, attribute_type: Type,
         infer = is_subtype(rvalue_type, get_type) and is_subtype(get_type, set_type)
         return rvalue_type if infer else set_type, get_type, infer
 
+    def check_parent_member_assignment(self, lvalue_type: Type,
+                                       lvalue_attribute_type: Type, rvalue_type: Type,
+                                       rvalue: Expression, context: MemberExpr) -> Type:
+        """Type inherited member assignment.
+
+        This defers to check_simple_assignment if the attribute is local to the instance_type,
+        otherwise it will inform the user where the attribute type was originally defined.
+
+        Return the inferred rvalue_type.
+        """
+        if isinstance(lvalue_type, (Instance)):
+            attribute_is_defined = lvalue_type.has_readable_member(context.name)
+            if isinstance(lvalue_type, UnionType):
+                direct_lvalue_vars = list(itertools.chain.from_iterable(
+                                          i.type.names for i in lvalue_type.items))
+                lvalue_bases = list(itertools.chain.from_iterable(
+                                    i.type.bases for i in lvalue_type.items))
+            else:
+                direct_lvalue_vars = lvalue_type.type.names
+                lvalue_bases = lvalue_type.type.bases
+                if '__metaclass__' in lvalue_type.type.names:
+                    # if the assigned metaclass is not class itself we have other problems
+                    metacls = lvalue_type.type.names['__metaclass__'].type.ret_type  # type: ignore
+                    lvalue_bases.append(metacls)
+                    attribute_is_defined = (attribute_is_defined or
+                                            metacls.has_readable_member(context.name))
+            if attribute_is_defined and context.name not in direct_lvalue_vars:
```

Comment:
Why do you need to check for `attribute_is_defined`, to avoid this extra message if the type is from a `__getattr__()`? Please add a comment. Naively, if an attribute is not defined at all, we can't know its type.